visualize-admin/visualization-tool
# perf: Only load hierarchies where necessary

Contributes to fixing #1002.

Previously, we loaded hierarchies every time we depended on `dimensionMetadata` fragment. However, I believe it's not necessary to load hierarchies everywhere, as for example, they don't seem to be used in places that use `DataCubeObservations` query. When we don't fetch them there, the loading times of the chart inside the editor & published mode can be improved for datasets that use hierarchies (in the example in #1002, the published chart loads in 1.5s instead of ~10s).

This PR excludes loading of hierarchies across multiple queries if they are not required further down in the app.

### How to test
1. Go to [this published chart](https://visualization-tool-git-perf-published-chart-ixt1.vercel.app/en/v/RF4kn1KqqdDN?dataSource=Test) from this PR's deployment preview
2. Go to the [same chart](https://test.visualize.admin.ch/en/v/tVsVHOmiGU3h?dataSource=Test) but made using test.visualize.admin.ch and see the time difference in data loading.
